AsterNET.ARI/Middleware/IEventProducer.cs

```diff
@@ -1,13 +1,8 @@
 ﻿using System;
+using WebSocket4Net;
 
 namespace AsterNET.ARI.Middleware
 {
-    public enum ConnectionState
```

Comment:
The middleware is now dependant on the WebSocket's enum, and not a Middleware provided one. This would mean any new middleware would need to reference the WebSocket without ever using it?! 

You should just have added the connecting state to the existing Enum.